### PR TITLE
added unit testing for ContextualizedClassifier models

### PR DESCRIPTION
**Pull request by juwaynilucman**
This is a follow-up on ContextualizedClassifier with BCELoss #140 

test_classifier checks:

- initial loss > final loss
- predicted parameters are not Nan

### CODE REVIEW

File: contextualized/easy/tests/test_classifier.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        y_dim = 1
+        C = np.random.uniform(-1, 1, size = (n_samples, c_dim))
+        X = np.random.uniform(-1, 1, size = (n_samples, x_dim))
+        Y = np.random.binomial(1, 0.5, size = (n_samples, y_dim))
+
+        model = ContextualizedClassifier(
+            alpha=1e-1, encoder_type='mlp'
+        )
+        self._quicktest(
+            model, C, X, Y, max_epochs=10
```
Comment (cnellington) [inline contextualized/easy/tests/test_classifier.py:50]: Has 10 epochs been enough in your experience to get NaN values from the old classifier?

**Review by cnellington — COMMENTED** [review]
LGTM! One comment to address.

File: contextualized/easy/tests/test_classifier.py (hunk 1 above)
Comment (juwaynilucman) [reply contextualized/easy/tests/test_classifier.py:50]: Using the sciPlex2 dataset, I used to get NaN values within the first epoch, so I assume 10 epochs will be enough for this test case.

Also, in my local machine, there are times when training in test_classifier.py and test_regressor.py ends before it reaches the assigned number of epochs. I am not sure why it happens, but could you confirm if it also happens in your machine?

Comment (cnellington) [reply contextualized/easy/tests/test_classifier.py:50]: Early stopping is enabled by default for the easy modules, but we should turn this off when testing the regression and classifier modules. Here's the relevant kwargs/callback constructor: https://github.com/cnellington/Contextualized/blob/30cb09c0261899584c61eb93997f1ca8a624ca3e/contextualized/easy/wrappers/SKLearnWrapper.py#L167
You should be able to disable this by setting es_patience to something like -1 or inf.

File: contextualized/easy/tests/test_classifier.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        X = np.random.uniform(-1, 1, size = (n_samples, x_dim))
+        Y = np.random.binomial(1, 0.5, size = (n_samples, y_dim))
+
+        model = ContextualizedClassifier(
+            alpha=1e-1, encoder_type='mlp'
+        )
+        self._quicktest(
+            model, C, X, Y, max_epochs=10, es_patience=float('inf')
+        )
+
```
Comment (cnellington) [inline contextualized/easy/tests/test_classifier.py:52]: Lots of whitespace, let's run black and make a new pylint badge for this update. Tests look great though.

**Review by cnellington — APPROVED** [review]
Looks great!
